octree/basic.go

```diff
@@ -131,8 +131,13 @@ func (octree *basicOctree) At(x, y, z float64) (pc.Data, bool) {
 	return nil, false
 }
 
-// Iterate TODO: Implement iterate for octree.
+// Iterate takes in the number of batches, a batch id, and a function to apply to each point in a basic octree
+// until false is returned. These parameters are passed to iterateRecursive along with the starting index (0)
+// to iterate through a basic octree if the number of batches is greater than or equal to 0.
 func (octree *basicOctree) Iterate(numBatches, myBatch int, fn func(p r3.Vector, d pc.Data) bool) {
+	if numBatches >= 0 {
```

Comment:
It should always be zero or positive if used as intended but shouldn't we also be checking to ensure numBatches is >= 0 since Iterate is exposed?  Technically someone could send a negative numBatches and 5%-2 does return a value (-1). According to the definition of Iterate only positive values should result in performing the batching process.

1. numBatch > 0: results in batching
2. numBatch == 0: results in no batching (we could error out if myBatch is non-zero or just ignore it and dont batch based on just the numBatch)
3. numBatch < 0: error out? Or ignore?

Reply:
I think checking if `numBatches < 0` and then erroring out makes the most sense.